### PR TITLE
refactor: add AbstractIcon class and make Icon extend it

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Create AbstractIcon class to enable adding new subclasses for SVG and Font icons

### CODE REVIEW

**Review by tomivirkki — APPROVED** [review]

File: vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.component.shared.HasTooltip;
+import com.vaadin.flow.dom.ElementConstants;
+
+/**
+ * Component for displaying an icon from the
+ * <a href="https://vaadin.com/icons">Vaadin Icons</a> collection.
```
Comment (tomivirkki) [inline vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java:29]: The description isn't correct

File: vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * @author Vaadin Ltd
+ * @see VaadinIcon
+ */
+@Tag("vaadin-icon")
+@NpmPackage(value = "@vaadin/polymer-legacy-adapter", version = "24.2.0-alpha11")
+@JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
+@NpmPackage(value = "@vaadin/icon", version = "24.2.0-alpha11")
+@JsModule("@vaadin/icon/src/vaadin-icon.js")
+public abstract class AbstractIcon extends Component
+        implements HasStyle, ClickNotifier<Icon>, HasTooltip {
```
Comment (tomivirkki) [inline vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java]: Screenshot 2023-08-28 at 16 10 31

HasStyle is redundant

File: vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    public abstract void setColor(String color);
+
+    /**
+     * Gets the color of this icon as a String.
+     *
+     * @return the color of the icon, or <code>null</code> if the color has not
+     *         been set
+     */
+    public abstract String getColor();
+}
```
Comment (tomivirkki) [inline vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java:79]: Needs a newline

File: vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java (hunk 1 above)
Comment (DiegoCardoso) [reply vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java]: Fixed.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**